pkg/kubernetes/node/v1alpha1/kubernetes.go

```diff
@@ -120,3 +172,53 @@ func (k *Kubeclient) List(opts metav1.ListOptions) (*corev1.NodeList, error) {
 	}
 	return k.list(cli, opts)
 }
+
+// Patch patches the Node in kubernetes cluster
+func (k *Kubeclient) Patch(name string, pt types.PatchType, data []byte, subresources ...string) (*corev1.Node, error) {
+	if len(name) == 0 {
+		return nil, errors.New("failed to patch node: missing node name")
+	}
+	cli, err := k.getClientsetOrCached()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to patch node : {%s}", name)
+	}
+	return k.patch(cli, name, pt, data, subresources...)
+}
+
+// Update updates the Node in kubernetes cluster
+func (k *Kubeclient) Update(node *corev1.Node) (*corev1.Node, error) {
+	if len(node.Name) == 0 {
+		return nil, errors.New("failed to patch node: missing node name")
+	}
+	cli, err := k.getClientsetOrCached()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to patch node : {%s}", node.Name)
+	}
+	return k.update(cli, node)
+}
+
+// PatchOrReplace uses given clientset to update the node status, either by patching or
+// updating the given node object; it may return error if the object cannot be encoded as
+// JSON, or if either patch or update calls fail; it will also return a second error
+// whenever creating a patch has failed
+func (k *Kubeclient) PatchOrReplace(node *corev1.Node, cordon bool) (error, error) {
```

Comment:
Let us have these methods:
```go
func (k *Kubeclient) Cordon(node *corev1.Node, isCordon bool) error
func (k *Kubeclient) CordonViaUpdate(node *corev1.Node, isCordon bool) error
func (k *Kubeclient) CordonViaPatch(node *corev1.Node, isCordon bool) error
```

Reply:
Then we can have just 2 methods `CordonViaUpdate` and `CordonViaPatch` .

Reply:
Cordon will make use of CordonViaUpdate

Reply:
Then how to switch the methods patch vs update ? if we don't check the errors.